git_machete/tests/functional/test_machete.py

```diff
@@ -2799,3 +2799,40 @@ def test_add(self) -> None:
         self.assert_command(['add', '--onto=feature'],
                             'Added branch `chore/remove_indentation` onto `feature`\n',
                             strip_indentation=False)
+
+    @mock.patch('git_machete.utils.run_cmd', mock_run_cmd)
+    def test_clean(self) -> None:
+        (
+            self.repo_sandbox.new_branch('master')
+                .commit()
+                .push()
+                .new_branch('bar')
+                .commit()
+                .new_branch('bar2')
+                .commit()
+                .push()
+                .check_out("master")
+                .new_branch('foo')
+                .commit()
+                .new_branch('foo2')
+                .commit()
+                .push()
+                .check_out("master")
+                .new_branch('moo')
+                .commit()
+                .new_branch('moo2')
+                .commit()
+        )
+        self.launch_command('discover', '-y')
```

Comment:
Is at least one untracked AND unmanaged branch removed in this test?

Reply:
I am not sure its possible to test deleting of unmanaged branch, same as there are no tests for `delete-unmanaged` command.

Reply:
It is indeed! just stop relying on `discover` for filling up .git/machete (coz `discover` will include **every** branch in .git/machete); instead fill up .git/machete directly, leaving out some branches as unmanaged

Reply:
I guess i badly explained what i meant, it is possible to have the unmanaged branch deleted but i couldn't find the way to test that it actually got deleted.

Reply:
UPDATE: I found the way, a good night sleep can do wonders :D